[ci] Use github actions for flake8

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25824 [ci] Use github actions for flake8**

Use github actions for flake8. This is nice because it makes it easier
to create inline annotations for lint violations.

It ends up looking like this:
https://github.com/suo/pytorch/pull/21/files

Differential Revision: [D17487007](https://our.internmc.facebook.com/intern/diff/D17487007)

suo committed:
[ci] Use github actions for flake8

Use github actions for flake8. This is nice because it makes it easier
to create inline annotations for lint violations.

It ends up looking like this:
https://github.com/suo/pytorch/pull/21/files

suo committed:
Update on "[wip][ci] Use github actions for flake8"


Use github actions for flake8. This is nice because it makes it easier
to create inline annotations for lint violations.

It ends up looking like this:
https://github.com/suo/pytorch/pull/21/files

suo committed:
Update on "[ci] Use github actions for flake8"


Use github actions for flake8. This is nice because it makes it easier
to create inline annotations for lint violations.

It ends up looking like this:
https://github.com/suo/pytorch/pull/21/files

suo committed:
Update on "[ci] Use github actions for flake8"


Use github actions for flake8. This is nice because it makes it easier
to create inline annotations for lint violations.

It ends up looking like this:
https://github.com/suo/pytorch/pull/21/files

ezyang commented on .github/workflows/lint.yml:
```diff
@@ -0,0 +1,29 @@
+          pip install flake8
+          flake8 --exit-zero > ${GITHUB_WORKSPACE}/flake8_output.txt
+      - name: Add annotations
+        uses: suo/add-annotations-github-action@master
```
Can we put this repo in pytorch org, managed by OSS, pretty please? :) Whitelist your new repo https://our.internmc.facebook.com/intern/diff/D16811608/

ezyang commented:
Consider deleting the old lint now?

suo commented:
Unfortunately this is not quite ready pending some questions on how Actions implements their webhook trigger :(

suo committed:
Update on "[ci] Use github actions for flake8"


Use github actions for flake8. This is nice because it makes it easier
to create inline annotations for lint violations.

It ends up looking like this:
https://github.com/suo/pytorch/pull/21/files

suo committed:
Update on "[ci] Use github actions for flake8"


Use github actions for flake8. This is nice because it makes it easier
to create inline annotations for lint violations.

It ends up looking like this:
https://github.com/suo/pytorch/pull/21/files

suo committed:
Update on "[ci] Use github actions for flake8"


Use github actions for flake8. This is nice because it makes it easier
to create inline annotations for lint violations.

It ends up looking like this:
https://github.com/suo/pytorch/pull/21/files
